### PR TITLE
changefeedccl: improve performance of gzip, and add zstd compression

### PR DESCRIPTION
**Pull request by miretskiy**
Expand the set of supported compression algorithms in changefeed.

A faster implementation of gzip algorithm is avaible, and is used
by default.  The gzip algorithm implementation can be reverted
to Go standard gzip implementation via the
`changefeed.fast_gzip.enabled` setting.

In addition, add support for compression files with zstd.

Addresses https://github.com/cockroachdb/cockroach/issues/88585

Release notes (enterprise change): Changefeed can emit files compressed
with zstd algorithm -- which provides good compression, and is much
faster than gzip.  In addition, a new, faster implementation of
gzip is used by default.

### CODE REVIEW

**Review by HonoreDB — APPROVED** [review]
> Reviewed 6 of 6 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88635)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stevendanna)

File: pkg/ccl/changefeedccl/compresion.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	switch algo {
+	case sinkCompressionGzip:
+		if useFastGzip.Get(sv) {
+			return pgzip.NewWriterLevel(dest, pgzip.DefaultCompression)
+		}
+		return stdgzip.NewWriterLevel(dest, stdgzip.DefaultCompression)
+	case sinkCompressionZstd:
+		return zstd.NewWriter(dest, zstd.WithEncoderLevel(zstd.SpeedFastest))
+	default:
+		return nil, errors.AssertionFailedf("unsupported encoder algorithm %q", algo)
```
Comment (stevendanna) [inline pkg/ccl/changefeedccl/compresion.go:57]: Perhaps?

```suggestion
		return nil, errors.AssertionFailedf("unsupported compression algorithm %q", algo)
```

**Review by stevendanna — APPROVED** [review]
LGTM.

**Comment by shermanCRL** [comment]
cc @nollenr

File: pkg/ccl/changefeedccl/compresion.go (hunk 1 above)
Comment (miretskiy) [reply pkg/ccl/changefeedccl/compresion.go:57]: Ack.

File: pkg/ccl/changefeedccl/compresion.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 The Cockroach Authors.
```
[the remark this answers is not in the document]
Comment (miretskiy) [reply pkg/ccl/changefeedccl/compresion.go:1]: Ohh.... Ooops... Or .. maybe it was compres'd?

**Comment by shermanCRL** [comment]
Backported to v22.2.1 via https://github.com/cockroachdb/cockroach/pull/91002

**Pull request merged**